### PR TITLE
feat: [CDS-40492]: Adding new end point and object

### PR DESCRIPTION
**Pull request by Y0709**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 8 Issues 
* Code issue 7 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37632)

File: 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV2.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import lombok.experimental.FieldDefaults;
+
+@OwnedBy(CDC)
+@Data
+@NoArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonDeserialize(using = JiraIssueCreateMetadataV2Deserializer.class)
+public class JiraIssueCreateMetadataNGV2 {
```
Comment (lasdox) [inline 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV2.java:40]: Lets rename this to a meaningful name, like: JiraIssueCreateMetaIssueTypes
something like this

File: 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV3.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import lombok.experimental.FieldDefaults;
+
+@OwnedBy(CDC)
+@Data
+@NoArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonDeserialize(using = JiraIssueCreateMetadataV3Deserializer.class)
+public class JiraIssueCreateMetadataNGV3 {
```
Comment (lasdox) [inline 970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV3.java:40]: Lets rename this to a meaningful name, like: JiraIssueCreateMetaFields
something like this

File: 120-ng-manager/src/main/java/io/harness/ng/jira/resources/JiraResource.java [hunk 3]
```diff
@@ -113,6 +115,38 @@ public ResponseDTO<JiraIssueCreateMetadataNG> getIssueCreateMetadata(
...
 
+  @GET
+  @Path("createMetadataV2")
+  @ApiOperation(value = "Get jira issue create metadata V2", nickname = "getJiraIssueCreateMetadataV2")
+  public ResponseDTO<JiraIssueCreateMetadataNGV2> getIssueCreateMetadataV2(
+      @NotNull @QueryParam("connectorRef") String jiraConnectorRef,
+      @NotNull @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
+      @QueryParam(NGCommonEntityConstants.ORG_KEY) String orgId,
+      @QueryParam(NGCommonEntityConstants.PROJECT_KEY) String projectId, @QueryParam("projectKey") String projectKey,
+      @QueryParam("issueTypeId") String issueTypeId, @QueryParam("fetchStatus") boolean fetchStatus,
```
Comment (lasdox) [inline 120-ng-manager/src/main/java/io/harness/ng/jira/resources/JiraResource.java:126]: we dont need issueTypeId here. lets try to clean these queryparams

**Comment by Y0709** [comment]
trigger ti0, ti1,ti2,ti3,ti4

**Comment by Y0709** [comment]
trigger ti1, ti4, ft1

**Review by prabu-kr — DISMISSED** [review]
> Reviewed 1 of 14 files at r2, 11 of 19 files at r3, 1 of 1 files at r4, 9 of 11 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: :shipit: complete! ✓

**Review by hingerabhinav — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: :shipit: complete! ✓

___
*[`956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java` line 453 at r6](https://reviewable.io/reviews/harness/harness-core/37632#-NDmEZG_6CA8NXo98MsV:-NDmEZG_6CA8NXo98MsW:b-inarcm) ([raw file](https://github.com/harness/harness-core/blob/157fd7005544a0f8c00064014fe277594e023342/956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java#L453)):*
> ```Java
>       "Enables the new way of how to copy powershell/winrm script commands content to file on remote. (Copy is done in chunks of 4KB) ",
>       HarnessTeam.CDP),
>   SERVICE_ENV_RECONCILIATION("Do reconciliation of service and env on pipeline/template save", HarnessTeam.CDC),
> ```

this is an extra FF

**Comment by Y0709** [comment]
trigger authorcheck

File: 960-api-services/src/test/java/io/harness/jira/JiraClientTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    assertThat(jiraIssueCreateMetadataNG.getProjects().size()).isEqualTo(1);
+    assertThat(jiraIssueCreateMetadataNG.getProjects().get("TES").getIssueTypes().size()).isEqualTo(1);
+    assertThat(jiraIssueCreateMetadataNG.getProjects().get("TES").getIssueTypes().get("Task").getId())
+        .isEqualTo("10001");
+    assertThat(jiraIssueCreateMetadataNG.getProjects().get("TES").getIssueTypes().get("Task").getFields().size())
+        .isEqualTo(6);
+    assertThat(jiraIssueCreateMetadataNG.getProjects().get("TES").getIssueTypes().get("Task").getFields().get("Labels"))
+        .isEqualTo(jiraFieldNG5);
+  }
+}
```
Comment (lasdox) [inline 960-api-services/src/test/java/io/harness/jira/JiraClientTest.java:124]: put a new line

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 14 files at r2, 6 of 19 files at r3, 1 of 1 files at r4, 8 of 11 files at r5, 3 of 5 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: :shipit: complete! ✓

**Comment by Y0709** [comment]
trigger authorcheck

**Review by Y0709 — COMMENTED** [review]
> Reviewed 1 of 14 files at r2, 6 of 19 files at r3, 1 of 1 files at r4, 8 of 11 files at r5, 3 of 5 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: 1. unresolved discussions. 

___
*[`956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java` line 453 at r6](https://reviewable.io/reviews/harness/harness-core/37632#-NDmEZG_6CA8NXo98MsV:-NDmucO_EfdPuF-GFsBa:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/157fd7005544a0f8c00064014fe277594e023342/956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java#L453)):*
<details><summary><i>Previously, hingerabhinav (Abhinav Hinger) wrote…</i></summary><blockquote>

this is an extra FF
</blockquote></details>

done

**Comment by abhinav-harness** [comment]
doing blind approve as everything is behind ff

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 14 files at r2, 6 of 19 files at r3, 1 of 1 files at r4, 8 of 11 files at r5, 3 of 5 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 19 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632#-:-NDnf9Aw5wr3qGBPG5rV:bnfp4nl)* status: 1. unresolved discussions.

**Comment by abhinav-harness** [comment]
trigger codebasehash

**Review by Y0709 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: 5. unresolved discussions. 

___
*[`960-api-services/src/test/java/io/harness/jira/JiraClientTest.java` line 124 at r8](https://reviewable.io/reviews/harness/harness-core/37632#-NDmr_zUnuPP0c9mxZjQ-r8-124:-NDnfLi76c8f6rwUc48c:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/39d32a4aa8080985c8e15e944005320951b2857b/960-api-services/src/test/java/io/harness/jira/JiraClientTest.java#L124)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

put a new line
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/jira/resources/JiraResource.java` line 126 at r2](https://reviewable.io/reviews/harness/harness-core/37632#-NCJZiUXjVmaypX2fH5y-r2-126:-NDnfQWtDOy_uM_lG-tb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/37585b4431986ad27847a1fa50d1478965201e60/120-ng-manager/src/main/java/io/harness/ng/jira/resources/JiraResource.java#L126)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

we dont need issueTypeId here. lets try to clean these queryparams
</blockquote></details>

Done.
___
*[`970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV2.java` line 40 at r2](https://reviewable.io/reviews/harness/harness-core/37632#-NCJZiUXjVmaypX2fH69-r2-40:-NDnfNF1CsZX5DmYf5Ym:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/37585b4431986ad27847a1fa50d1478965201e60/970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV2.java#L40)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Lets rename this to a meaningful name, like: JiraIssueCreateMetaIssueTypes
something like this
</blockquote></details>

Done.
___
*[`970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV3.java` line 40 at r2](https://reviewable.io/reviews/harness/harness-core/37632#-NCtXJSMylIoSXexyJ5n-r2-40:-NDnfMh6Ct9xl_HQB1zK:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/37585b4431986ad27847a1fa50d1478965201e60/970-api-services-beans/src/main/java/io/harness/jira/JiraIssueCreateMetadataNGV3.java#L40)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Lets rename this to a meaningful name, like: JiraIssueCreateMetaFields
something like this
</blockquote></details>

Done.

**Review by Y0709 — COMMENTED** [review]
> Reviewed 4 of 19 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37632)* status: 5. unresolved discussions.

**Pull request merged**